directives/point_of_interest.py

```diff
@@ -32,50 +40,106 @@ class PointOfInterest(Directive):
     has_content = True
     final_argument_whitespace = True
     option_spec = {
+        'id': directives.unchanged,
         'title': directives.unchanged,
         'previous': directives.unchanged,
         'next': directives.unchanged,
         'hidden': directives.flag,
         'class' : directives.class_option,
+        'height': directives.length_or_percentage_or_unitless,
+        'columns': directives.unchanged,
+        'bgimg':directives.uri,
     }
 
     def run(self):
         self.assert_has_content()
+        env = self.state.document.settings.env
         if len(self.arguments) == 0:
             return [nodes.container()]
-        name = self.arguments[0]
-        title_text = ''
-        if 'title' in  self.options:
+
+        if 'title' in self.options:
+            if 'id' in self.options:
+                raise SphinxError(u'Point of interest options can\'t contain both "title" and "id"; one of them should be provided as an argument instead')
+            name = self.arguments[0]
             title_text = self.options['title']
+        else:
+            name = self.options.get('id', ''.join(random.choice(string.ascii_lowercase + string.digits) for i in range(6)))
+            title_text = self.arguments[0]
```

Comment:
Does this handle spaces in the string? e.g.

```
.. point-of-interest:: My long title with multiple words
```

Reply:
Yes, spaces in the title are fine.

Reply:
Good. I didn't remember how the arguments are split by sphinx.